nucypher/blockchain/eth/actors.py

```diff
@@ -787,14 +788,29 @@ def _divide_stake(self, stake_index: int, additional_periods: int, target_value:
 
     def _deposit_and_increase(self, stake_index: int, amount: int) -> TxReceipt:
         """Public facing method for deposit and increasing stake."""
-        self.token_agent.increase_allowance(increase=amount,
-                                            transacting_power=self.transacting_power,
-                                            spender_address=self.staking_agent.contract.address)
+        self._ensure_allowance_equals(amount)
         receipt = self.staking_agent.deposit_and_increase(transacting_power=self.transacting_power,
                                                           stake_index=stake_index,
                                                           amount=amount)
         return receipt
 
+    def _ensure_allowance_equals(self, amount: int):
+        owner = self.checksum_address
```

Comment:
I lost track about the invariant around `self.checksum_address`. Is it always the case that this address is the same than the transacting one? Just in case, I would do something like:
```suggestion
        owner = self.transacting_power.account
```